File: ExampleXOR.R
source("Network.R")
source("FCLayer.R")
source("ActivationLayer.R")
source("Activation.R")
source("LossFunction.R")

# training data
x_train <- matrix(rbind(c(0,0),c(0,1),c(1,0),c(1,1)),4,2)
y_train <- matrix(c(0,1,1,0), 4, 1)

# network
net = Network()
net$add(FCLayer(2, 3))
net$add(ActivationLayer(relu, relu_prime))
net$add(FCLayer(3, 1))
net$add(ActivationLayer(relu, relu_prime))

# train
net$use(mse, mse_prime)
net$fit(x_train, y_train, epochs=1000, learning_rate=0.1)

# test
out <- net$predict(x_train)
print(out)

File: ExampleXOR.jl
include("FCLayer.jl")

x = [0 0; 0 1; 1 0; 1 1]
y = [1.0; 0.0; 0.0; 1.0][:, :]

L1 = FCLayer(2, 3)
L2 = ActivationLayer(L1, tgh, tgh_prime)
L3 = FCLayer(3, 1)
L4 = ActivationLayer(L3, tgh, tgh_prime)
layers = [L1 L2 L3 L4]

fit(layers, x, y, 0.5, 1000)
out = predict(layers, x)
print(out)
